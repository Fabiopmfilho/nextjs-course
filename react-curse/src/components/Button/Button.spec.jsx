import { fireEvent, render, screen } from "@testing-library/react";
import { Button } from ".";

describe("<Button />", () => {
  test('should render the button with the text "Load more"', () => {
    render(<Button text="Load more posts"/>);

    expect.assertions(1)

    const button = screen.getByRole("button", { name: /load more posts/i });
    expect(button).toBeInTheDocument();
  });
  
  test('should call function on button click', () => {
    const fn = jest.fn()
    render(<Button text="Load more posts" onClick={fn} />);

    const button = screen.getByRole("button", { name: /load more posts/i });
    
    fireEvent.click(button);

    expect(fn).toHaveBeenCalled()
  });
  
  test('should be disabled when disabled is true', () => {
    render(<Button text="Load more posts" disabled={true} />);

    const button = screen.getByRole("button", { name: /load more posts/i });
    
    expect(button).toBeDisabled();
  });
  
  test('should be enabled when disabled is false', () => {
    render(<Button text="Load more posts" disabled={false} />);

    const button = screen.getByRole("button", { name: /load more posts/i });
    
    expect(button).toBeEnabled();
  });
  
  test('should be enabled when disabled is false', () => {
    const fn = jest.fn()
    render(<Button text="Load more posts" disabled={false} />);

    const {container} = render(<Button text="Load more posts" disabled={false} />);
    
    expect(container.firstChild).toMatchSnapshot();
  });
});
