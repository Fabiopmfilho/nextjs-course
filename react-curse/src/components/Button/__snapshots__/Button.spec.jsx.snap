// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be enabled when disabled is false 1`] = `
<button
  class="button"
  placeholder="Type your search"
>
  Load more posts
</button>
`;
